### DISCUSSION TITLE
Update "Fundamentals" lesson structure

### DISCUSSION THREAD

**Discussion by didoesdigital** [didoesdigital/typey-type#84] [Building Typey Type]
# Overview

The original Typey Type lesson structure followed the [Learn Plover book lesson structure](https://sites.google.com/site/learnplover/home). There might be better options for teaching Plover/steno fundamentals.

## Background

From the [README](https://github.com/didoesdigital/typey-type#readme)

> The aim of Typey Type is to fast-track learning and mastering steno skills. Typey Type helps novice stenographers by giving them:
    - immediate feedback on typing speed and accuracy
    - progress updates, including progress on brief vocabulary
    - a course to follow, including material to type
    - links to more information

> Typey Type concentrates on practical skills, such as physical drilling and memorisation of briefs.

> Typey Type is not a generic typing app. Plenty of typing apps already exist for regular text input without stenography.

> Instead, Typey Type concentrates on teaching people how to type using stenography. That means it includes features like steno diagrams showing steno briefs as hints for how to write words and using steno-specific data, such as lessons for “single-stroke briefs”.

The aim of the [steno dictionaries that power Typey Type](https://github.com/didoesdigital/steno-dictionaries) is to remove all the misstrokes from `dict.json` to give new stenographers greater confidence in learning new briefs. The first step is to remove all the misstrokes for the _shortest_ available brief for every word, starting with words appearing in Typey Type's top 10,000 words lessons.

## Considerations

On the material itself:

- Typey Type is "brief and one-stroker heavy, so a lot of learners are learning one-stroke outlines even before they've learnt the easier phonetic ones"
- Typey Type material might be "too much at once and might actually be scaring people away"
- "things that are under the Fundamentals category are actually fairly advanced techniques that shouldn't be taught until the user has had enough practice with syllable-by-syllable word entry without inversion"
- Relevant historical thread on brief selection for showing hints: https://github.com/didoesdigital/steno-dictionaries/issues/41

On making changes:

- Maintaining existing links/lessons (e.g. adding additional lessons in between might help keep links on the Internet functioning reliably versus changing the existing lessons)
- Change management for existing users e.g. for someone that is "half way through the lessons", we might avoid pulling the rug out from under them by adding lots of new lessons
- Fine-tuning the criteria for each lesson might reduce lesson length as a quick win without disrupting a lot
- The existing static lesson generator uses criteria like "one syllable", "multi-syllable", "has inversion", "has right-hand side consonant keys", "has compound clusters", "uses F as V or S" and lesson definitions like the "multi-syllable words with suffixes lesson" is made of words that use "suffixes", "more than one syllable", and don't contain an asterisk. It would be good to define new lessons in similar terms.
    - Di has general plans to change how lessons are built/maintained so that you can create a lesson using a word list (e.g. "UX design words") or criteria/rules (e.g. "one syllable, no inversion, at least 5 characters", etc.) from a corpus (e.g. top 10000 project Gutenberg words, whole Plover dictionary, etc.); see [related issue](https://github.com/didoesdigital/typey-type-data/issues/1)
- Could/should we support multiple lesson plans e.g. for people optimising for ergonomics, do these lessons in this order; for people using X theory, do these lessons in this order

**Comment by didoesdigital** [comment]
I'd like to add a "Day 1" lesson before "Introduction" that can be done in less than, say, 10 minutes at 3 words per minute, and contains a small number of simple words that only use 4 or fewer keys (so that QWERTY folk with non-NKRO keyboards can try it, and live user testing shows that suddenly pressing 5 keys at once is daunting and physically difficult to do initially). Not sure exactly which words to use in this lesson.

**Reply by lloy0076** [reply]
Also, it's useful to give learners a sense of achievement - even 3 WPM is better than 0 WPM; the longer lessons were daunting to me at first (typed using my QWERTY or this issue will have been 500 years old before I got it out...)
